src/crypto/crypto_context.cc

```diff
@@ -188,28 +189,66 @@ int SSL_CTX_use_certificate_chain(SSL_CTX* ctx,
                                        issuer);
 }
 
+unsigned long LoadCertsFromFile(  // NOLINT(runtime/int)
+    std::vector<X509*>* certs,
+    const char* file) {
+  ClearErrorOnReturn clear_error_on_return;
```

Comment:
This clears openssl's _entire_ error stack instead of just the errors this function generates (compare `MarkPopErrorOnReturn`.)

I assume that's not what you want but if it is, can you explain why?

Reply:
The existing `AddCertsFromFile` function already clears all errors when it's called:
```
  ERR_clear_error();
  MarkPopErrorOnReturn mark_pop_error_on_return;
```

Using `MarkPopErrorOnReturn` here seemed strange because there the errors are cleared immediately before the call to `ERR_set_mark()`. I swapped it over to `ClearErrorOnReturn` to clearer, and left the clear because that's what the existing code used.

That said, `AddCertsFromFile` was called at startup and now `LoadCertsFromFile` can occur at any point when a secure context is created. Should I swap it to `MarkPopErrorOnReturn` without a clear as a precaution?

Reply:
Yes, probably. `ClearErrorOnReturn` gets overused in src/crypto due to a fluke of history but it's in fact rarely appropriate.

Reply:
Changes committed. Swapped to `MarkPopErrorOnReturn` and `ERR_peek_last_error` to avoid impacting any previous errors on the stack.